qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -170,8 +168,9 @@ def _build(self):
                 if _is_pauli_identity(op):
                     continue
 
-                evolved_op = self.evolution.convert((coeff * op).exp_i()).reduce()
-                circuits.append(evolved_op.to_circuit())
+                if op is not None:
+                    evolved_op = self.evolution.convert((coeff * op).exp_i()).reduce()
+                    circuits.append(evolved_op.to_circuit())
```

Comment:
Why is this new if-clause necessary? Is there a case where `op` is `None`?

Reply:
The answer is yes, but since I haven't looked at this in a couple of months, I'll have to rerun some tests and see why that is.